geoh5py/data/colour.py

```diff
@@ -0,0 +1,172 @@
+        """
+        Convert a 2D numpy array to RGB.
+
+        :param values: 2D numpy array with 3 or 4 bands.
+
+        :return: A structured array with "r", "g", "b" bands
+        """
+        if values.ndim != 2 or values.shape[1] not in (3, 4):
+            raise ValueError("Values must be a 2D numpy array containing RGB bands.")
+
+        if values.dtype != np.uint8:
+            values = min_max_scaler(values, 0, 255, axis=0)
+
+        if values.shape[1] == 3:
+            values = np.column_stack(
+                (values, np.full(values.shape[0], 255, dtype=np.uint8))
+            )
+
+        values = (
+            values[:, :4]
+            .astype(np.uint8)
+            .view([("r", "u1"), ("g", "u1"), ("b", "u1"), ("a", "u1")])
+            .reshape(-1)
+        )
+
+        return values
+
+    @staticmethod
+    def _structured_to_rgb(values: np.ndarray) -> np.ndarray:
+        """
+        Convert a structured array to RGB.
+
+        :param values: The structured array with "r", "g", "b" bands.
+
+        :return: The RGB values.
+        """
+
+        # convert the dtype names to lower case
+        values = values.astype(
+            [(name.lower(), values.dtype[name]) for name in values.dtype.names]
+        )
+
+        for band in ["r", "g", "b", "a"]:
+            if band not in values.dtype.names:
+                if band == "a":
+                    values = rfn.append_fields(
+                        values,
+                        names="a",
+                        data=np.full(values.shape[0], 255, dtype=np.uint8),
+                        usemask=False,
+                    )
+                else:
+                    raise ValueError(
+                        "Values must be a 2D numpy array containing RGB bands."
+                    )
+
+            # ensure the min and the max are between 0 and 255
+            if values[band].dtype != np.uint8:
+                values[band] = min_max_scaler(values[band], 0, 255).astype(np.uint8)
+
+        return values[["r", "g", "b", "a"]]
+
+    @classmethod
+    def _verify_parents(cls, parent: Any):
+        """
+        Verify if the parent is allowed for Colour data.
+
+        For now the objects that can use Colour data are limited.
+        Raises an error if the parent is not allowed.
+
+        :param parent: The parent object class.
+        """
+        if parent is not None:
+            # use name to avoid circular import
+            parent_classes = parent.__class__.__name__
+            if parent_classes not in cls.allowed_parent:
+                raise TypeError(
+                    f"Parent '{parent_classes}' is not allowed for Colour data.\n"
+                    f"The allowed parents are {cls.allowed_parent}"
+                )
+
+    @property
+    def nan_value(self):
+        return np.array(
+            [
+                (90, 90, 90, 0),
+            ],
+            dtype=[("r", "u1"), ("g", "u1"), ("b", "u1"), ("a", "u1")],
```

Comment:
Can we store that information as class attribute, like is done elsewhere such as: https://github.com/MiraGeoscience/geoh5py/blob/07722a173159bb412c664e07d4e793743adaec81/geoh5py/data/color_map.py#L40